app/src/common/shared/com/igalia/wolvic/ui/views/TabsBarItem.java

```diff
@@ -160,6 +158,17 @@ public void onLocationChange(@NonNull WSession session, @Nullable String url) {
         }
     }
 
+    private String getTitleForDisplay(String url, String title) {
+        Windows.ContentType contentType = UrlUtils.getContentType(url);
+        if (contentType != Windows.ContentType.WEB_CONTENT) {
```

Comment:
I'm not sure I understand this condition; it seems more like the "fallback" case, isn't it ? Wouldn't be clearer if we do the following instead ? 

```
if (contentType == Windows.ContentType.WEB_CONTENT)
  return !StringsUtils.isEmpty(title) ? title : UrlUtils.stripCommonSubdomains(UrlUtils.getHost(url));
  
return  getContext().getString(contentType.titleResId);
```

Reply:
Good suggestion, I've updated the PR.